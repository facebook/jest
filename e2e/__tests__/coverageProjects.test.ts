/**
 * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import * as fs from 'fs';
import * as path from 'path';
import {wrap} from 'jest-snapshot-serializer-raw';
import {run} from '../Utils';
import runJest from '../runJest';

const DIR = path.resolve(__dirname, '../coverage-projects');

beforeAll(() => {
  run('yarn', DIR);
});

test('outputs coverage report', () => {
  const {stdout, exitCode} = runJest(DIR, ['--no-cache', '--coverage'], {
    stripAnsi: true,
  });
  const coverageDir = path.join(DIR, 'coverage');

  // - the `index.js` file in package `mul` is ignored and should not be in the
  //  coverage report.
  expect(wrap(stdout)).toMatchSnapshot();

  // `index.js` should only appear once (in package `add`)
  expect(stdout.match(/index\.js/g)).toHaveLength(1);

  expect(() => fs.accessSync(coverageDir, fs.F_OK)).not.toThrow();
  expect(exitCode).toBe(0);
});
