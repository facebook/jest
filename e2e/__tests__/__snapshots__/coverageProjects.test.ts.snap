// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`outputs coverage report 1`] = `
----------------|---------|----------|---------|---------|-------------------
File            | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
----------------|---------|----------|---------|---------|-------------------
All files       |     100 |      100 |     100 |     100 |                   
 add            |     100 |      100 |     100 |     100 |                   
  index.js      |     100 |      100 |     100 |     100 |                   
 mul            |     100 |      100 |     100 |     100 |                   
  other_file.js |     100 |      100 |     100 |     100 |                   
----------------|---------|----------|---------|---------|-------------------
`;
